openmc/plotter.py

```diff
@@ -0,0 +1,910 @@
+            ylabel = 'Nuclidic Microscopic Data'
+        elif data_type == 'element':
+            ylabel = 'Elemental Microscopic Data'
+        elif data_type == 'material' or data_type == 'macroscopic':
+            ylabel = 'Macroscopic Data'
+    else:
+        if data_type == 'nuclide':
+            ylabel = 'Microscopic Cross Section [b]'
+        elif data_type == 'element':
+            ylabel = 'Elemental Cross Section [b]'
+        elif data_type == 'material' or data_type == 'macroscopic':
+            ylabel = 'Macroscopic Cross Section [1/cm]'
+    ax.set_ylabel(ylabel)
+    ax.legend(loc='best')
+    name = this.name if data_type == 'material' else this
+    if len(types) > 1:
+        ax.set_title('Cross Sections for ' + name)
+    else:
+        ax.set_title('Cross Section for ' + name)
+
+    return fig
+
+
+def calculate_cexs(this, data_type, types, temperature=294., sab_name=None,
+                   cross_sections=None, enrichment=None):
+    """Calculates continuous-energy cross sections of a requested type.
+
+    Parameters
+    ----------
+    this : {str, openmc.Nuclide, openmc.Element, openmc.Material}
+        Object to source data from
+    data_type : {'nuclide', 'element', material'}
+        Type of object to plot
+    types : Iterable of values of PLOT_TYPES
+        The type of cross sections to calculate
+    temperature : float, optional
+        Temperature in Kelvin to plot. If not specified, a default
+        temperature of 294K will be plotted. Note that the nearest
+        temperature in the library for each nuclide will be used as opposed
+        to using any interpolation.
+    sab_name : str, optional
+        Name of S(a,b) library to apply to MT=2 data when applicable.
+    cross_sections : str, optional
+        Location of cross_sections.xml file. Default is None.
+    enrichment : float, optional
+        Enrichment for U235 in weight percent. For example, input 4.95 for
+        4.95 weight percent enriched U. Default is None
+        (natural composition).
+
+    Returns
+    -------
+    energy_grid : numpy.ndarray
+        Energies at which cross sections are calculated, in units of eV
+    data : numpy.ndarray
+        Cross sections calculated at the energy grid described by energy_grid
+
+    """
+
+    # Check types
+    cv.check_type('temperature', temperature, Real)
+    if sab_name:
+        cv.check_type('sab_name', sab_name, str)
+    if enrichment:
+        cv.check_type('enrichment', enrichment, Real)
+
+    if data_type == 'nuclide':
+        if isinstance(this, str):
+            nuc = openmc.Nuclide(this)
+        else:
+            nuc = this
+        energy_grid, xs = _calculate_cexs_nuclide(nuc, types, temperature,
+                                                  sab_name, cross_sections)
+        # Convert xs (Iterable of Callable) to a grid of cross section values
+        # calculated on @ the points in energy_grid for consistency with the
```

Comment:
Question: Did you mean to say "on @ the points" here, or was this meant to be "@ the points" or "on the points"?